src/cmd/singularity/cli/sif.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/singularityware/singularity/docs"
+	"github.com/singularityware/singularity/src/pkg/buildcfg"
+	"github.com/singularityware/singularity/src/pkg/sylog"
+	"github.com/spf13/cobra"
+)
+
+var (
+	deffile  string
+	partfile string
+	content  string
+	fstype   string
+	parttype string
+	uuid     string
+)
+
+func init() {
+	SifCmd.Flags().SetInterspersed(false)
+	SingularityCmd.AddCommand(SifCmd)
+	// Create
+	SifCmd.AddCommand(SifCreate)
+	SifCreate.Flags().StringVarP(&deffile, "deffile", "D", "", "include definitions file 'deffile'")
+	SifCreate.Flags().StringVarP(&partfile, "partfile", "P", "", "include file system partition `partfile'")
+	SifCreate.Flags().StringVarP(&content, "CONTENT", "c", "", "freeform partition content string")
+	SifCreate.Flags().StringVarP(&fstype, "FSTYPE", "f", "", "filesystem type: EXT3, SQUASHFS")
+	SifCreate.Flags().StringVarP(&parttype, "PARTTYPE", "p", "", "filesystem partition type: SYSTEM, DATA, OVERLAY")
+	SifCreate.Flags().StringVarP(&uuid, "uuid", "u", "", "pass a uuid to use instead of generating a new one")
+	// List
+	SifCmd.AddCommand(SifList)
+	// Dump
+	SifCmd.AddCommand(SifDump)
+	// Header ifHeader
+	SifCmd.AddCommand(SifHeader)
+	// Info
+	SifCmd.AddCommand(SifInfo)
+	// Del
+	SifCmd.AddCommand(SifDel)
+}
+
+var sif = buildcfg.SBINDIR + "/sif"
+
+// SifCmd represent the sif CLI cmd
+var SifCmd = &cobra.Command{
+	Use:   docs.SifUse,
+	Short: docs.SifShort,
```

Comment:
I really think  these commands should all have `Long`. If there's a good reason not to, fair enough... but it looks to me like the other commands all do that, and it makes the help more readable.

Reply:
@tri-adam , I agree that all the commands and subcommands *should* have a long, but right now the standalone `sif` C binary does not have much information for @ArangoGutierrez to write a `Long` string for the subcommands. I propose that we merge in the current state and @ArangoGutierrez works with @yhcote to circle back on developing `Long` strings for the subcommands.